src/lib.rs

```diff
@@ -260,6 +260,16 @@ where
                 }
                 formatter.write_str("---")
             }
+            Math(ref math_display, ref math) => {
+                let delimiter = match math_display {
+                    MathDisplay::Inline => "$",
+                    MathDisplay::Block => "$$\n",
+                };
+                formatter.write_str(delimiter)?;
```

Comment:
Tests using both delimiters would be good to see. I also wonder if round-trips works consistently with the newline added unconditionally in the `MathDisplay::Block` case, or if there is anything else that's subtly breaking when the conditions are right.